src/pages/demo/custom-controls/index.md

```diff
@@ -0,0 +1,37 @@
+---
+date: "2015-03-10"
+title: "Custom controls"
+description: "Using the custom controls api"
+thumbnail: "custom-controls.png"
+tags: ['controls','customization']
+---
+
+This demo shows how to start using the **custom controls** api. This first example show how to write a custom render method for your control and how to define a custom action to run on the control mouseup. The new api offers the ability to run custom actions on:
+
+- mousedown
+- mouseup
+
+And to run custom code for
+
+- Positioning
+- Rendering
+- Cursor styling
+
+## Creating the control
+We create a new control and we add it to the object. To create a control we use `new fabric.Control(options)` In this case we want the corner to be near the top right classic control, so we give it a position of x of 0.5 that is the extreme right of the bounding box of the object. Same for y at -0.5. We do not want it to overlap with the current top right corner, so we add an `offsetY` with a positive number that will make it render a bit more down
+
+We want our control to render an icon rather than a fabric supported circle or square. So we have to provide our own render function to the control. The function is receiving the canvas context as first argument and so a simple drawImage will make it
+
+For the custom action we want simply to delete the object. We build a function following the mouseUpHandler signature and we use there `canvas.remove(target)`
+
+
```

Comment:
Here we can add the **CodePens** by embedding them in an iFrame as documented on [CodePen](https://blog.codepen.io/documentation/embedded-pens/)
Need to check the pricing caveats though, or maybe we can use offerings from other vendors.

Reply:
There is prefill embeds that is already working on the other website. So it will work here too.

Reply:
> You seems to have gigantic tabs. Would be possible to later to use the double space like in fabric.js?
Yes, I too noticed the TABS being gigantic! 😅

Sure, I cud switch to using double spaces, although I had been using TAB coz it helps configure the text editor to display tabs of desired width (either 2 or 4) when working on different screens!

Would check with a `.editorconfig` file id it helps! 😇